psi4/src/psi4/libfock/snLinK.cc

```diff
@@ -0,0 +1,494 @@
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#include "jk.h"
+#include "SplitJK.h"
+#include "psi4/libqt/qt.h"
+#include "psi4/libfock/cubature.h"
+#include "psi4/libfock/points.h"
+#include "psi4/libmints/vector.h"
+#include "psi4/libmints/basisset.h"
+#include "psi4/libmints/electrostatic.h"
+#include "psi4/libmints/mintshelper.h"
+#include "psi4/libmints/molecule.h"
+#include "psi4/libmints/integral.h"
+#include "psi4/libmints/petitelist.h"
+#include "psi4/liboptions/liboptions.h"
+#include "psi4/lib3index/dftensor.h"
+#include "psi4/libpsi4util/PsiOutStream.h"
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <tuple>
+#include <unordered_set>
+#include <variant>
+#include <vector>
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+using namespace psi;
+
+namespace psi {
+
+// creates maps for translating Psi4 option inputs to GauXC enums
+std::tuple<
+    std::unordered_map<std::string, GauXC::PruningScheme>, 
+    std::unordered_map<std::string, GauXC::RadialQuad> 
+> snLinK::generate_enum_mappings() {
+    // generate map for grid pruning schemes 
+    std::unordered_map<std::string, GauXC::PruningScheme> pruning_scheme_map; 
+    pruning_scheme_map["ROBUST"] = GauXC::PruningScheme::Robust;
+    pruning_scheme_map["TREUTLER"] = GauXC::PruningScheme::Treutler;
+    pruning_scheme_map["NONE"] = GauXC::PruningScheme::Unpruned;
+
+    // generate map for radial quadrature schemes 
+    std::unordered_map<std::string, GauXC::RadialQuad> radial_scheme_map; 
+    radial_scheme_map["TREUTLER"] = GauXC::RadialQuad::TreutlerAldrichs;
+    radial_scheme_map["MURA"] = GauXC::RadialQuad::MuraKnowles;
+    // The Murray, Handy, Laming literature reference is mentioned in cubature.cc
+    // with association to this keyword
+    radial_scheme_map["EM"] = GauXC::RadialQuad::MurrayHandyLaming; 
+
+    // we are done
+    return std::make_tuple(pruning_scheme_map, radial_scheme_map);
+}
+
+// constructs a permutation matrix for converting matrices to and from GauXC's integral ordering standard 
+Eigen::PermutationMatrix<Eigen::Dynamic, Eigen::Dynamic> snLinK::generate_permutation_matrix(const std::shared_ptr<BasisSet> psi4_basisset) {
+  
+    Eigen::PermutationMatrix<Eigen::Dynamic, Eigen::Dynamic> permutation_matrix(psi4_basisset->nbf());
+    
+    // general array for how to reorder integrals 
+    constexpr int max_am = GAUXC_MAX_AM;
```

Comment:
This is a compile time constant, which makes the code dependent on a specific binary of GauXC.

Reply:
We don't have a runtime checker for this yet, but it can be added. Tracking here
https://github.com/wavefunction91/GauXC/issues/122

Reply:
Thanks for the issue update! I'll update the code with runtime checking once the issue there is resolved. I suppose in the mean time, the best thing we can do is leave things here as is?